### DISCUSSION TITLE
What do the fastest way for loading massive data into the list?

### DISCUSSION THREAD

**Discussion by MohMehrnia** [ravendb/ravendb#14058] [General]
What do the fastest way for loading massive data into the list? (3600 Documents)? (Load in One time)

**Comment by ayende** [comment]
What do you mean, `the list` ? 
Do you mean reading thousands of documents into the session? Can you give more context?

What is the average size of those documents? How do you load them (by id, using a query, etc)?

**Reply by MohMehrnia** [reply]
3600 document with 2k for each document and I used query

**Reply by mcrio** [reply]
Check the streaming functionality: https://ravendb.net/docs/article-page/5.3/csharp/client-api/session/querying/how-to-stream-query-results

**Comment by ayende** [comment]
As @mcrio mentioned, the right way to do something like that is to use streaming.
That would be the fastest and most efficient manner to do this, since it avoids materializing the whole query result ~7GB before handing you the results.

See: https://ravendb.net/docs/article-page/5.3/csharp/client-api/session/querying/how-to-stream-query-results

As a question, what _do_ you need with 7GB of docs?
